### ISSUE TITLE
"excludeFields" is applied to all fields and not just to the fields in "Library"

### ISSUE THREAD

**Issue by SageDroid** [mikepenz/AboutLibraries#1014]
## About this issue

The documentation describes `excludeFields` as follows:

_Defines fields which will be excluded during the serialisation of the metadata output file.
Any field as included in the [com.mikepenz.aboutlibraries.plugin.mapping.Library] can theoretically be excluded._

**Problem:** `excludeFields` is applied to all fields and not just to the fields in `Library`. 

**Example:** If you exclude `name` (for `Library.name`), the `License.name`, `Developer.name` and `Organization.name` properties are also removed from the output. This means that it is no longer possible to parse the libraries and licenses at runtime, as `License.name` and `Organization.name` are not optional.

## Details
- [11.2.2] Used library version

## Checklist

- [x] Searched for [similar issues](https://github.com/mikepenz/AboutLibraries/issues)
- [x] Checked out the [sample application](https://github.com/mikepenz/AboutLibraries/tree/develop/app)
- [x] Read the [README](https://github.com/mikepenz/AboutLibraries/blob/develop/README.md)
- [x] Checked out the [CHANGELOG](https://github.com/mikepenz/AboutLibraries/releases)
- [x] Read the [MIGRATION GUIDE](https://github.com/mikepenz/AboutLibraries/blob/develop/MIGRATION.md)

**Comment by mikepenz** [comment]
Thank you for the ticket. 

I believe we should make sure not to break the current behavior for users. 
It probably would be better if we add new configuration options for the adjusted behavior.

**Comment by SageDroid** [comment]
Thanks for your reply (and this brilliant library)!

I generally agree with you. The changes in the pull request were based on my understanding of the documentation. I did not assume that the current behavior was intended.

Further configuration options would make using the library more complex. Especially if the old behavior and the new one exist in parallel.

One idea that just popped into my head is that we could allow to qualify the field name in `excludeFields` by the class name (e.g. `Library.name`). Unqualified field names would continue to be applied globally. This would offer full configurability and nobody would have to change their existing configuration. 

What do you think?

**Comment by mikepenz** [comment]
The later suggestion, using fully qualified field names is a great and would in my opinion tick all boxes. it won't break existing implementations, but also provide full flexibility for other usecases. 👍

**Comment by SageDroid** [comment]
I have updated my PR. The current behavior is still supported. In addition, it is now possible to specify fully qualified field names. In my example, that would be:

`excludeFields = ["com.mikepenz.aboutlibraries.plugin.mapping.Library.name"]`

I can also update the documentation (and possibly the readme - or do you do that with the release?) if you are okay with the new implementation.

**Comment by SageDroid** [comment]
Sorry I understood your previous reply to mean that you prefer fully qualified names.

I have updated the PR. The added list of allowed qualifiers (class names), ensures that it is still possible to exclude field names with dots (may occur in the licenses). All class names that are currently used in `ResultContainer` are permitted. I have refrained from building the list with reflection in order to keep the code easy to understand.

It's now `excludeFields = ["Library.name"]`

Is this solution ok for you?
